### PR TITLE
[IMP] open_academy: Relations between course and session T#59081

### PR DESCRIPTION
**Pull request by JesusValdez96**
- Add next relation fields as in the exercise "Many2one relations":
- Add Many2one relation as "course" field between session and course.
- Add Many2one field in session model to add a instructor field related to partners model.
- Add Many2one relation in course model with users to add responsible user field.

- Add One2many field "sessions" to reflect the relation between courses and sessions as exercise "Inverse one2many relations" indicates.

- Add Attendees Many2many field at session model as exercise "Multiple many2many relations".

- Add Responsible user field to Tree and Form views of course view.
- Add new fields to Session Tree view.
- Add Session Form view.

Responsible user in Course Tree view:
responsible field

Responsible user and Sessions in Course Form view:
session field in course

Session Tree view:
session tree view

Session Form view:
session form view

Link to exercises: https://www.odoo.com/documentation/15.0/developer/howtos/backend.html#relational-fields

### CODE REVIEW

**Comment by JesusValdez96** [comment]
Please review @andreaarce, @CarmenMiranda

File: open_academy/models/course.py [hunk 1]
```diff
@@ -5,5 +5,7 @@ class Course(models.Model):
     _name = 'course'
     _description = 'Open Academy Course'
 
-    title = fields.Char(required=True)
+    name = fields.Char(required=True)
     description = fields.Text()
+    responsible_user = fields.Many2one("res.users")
```
Comment (CarmenMiranda) [inline open_academy/models/course.py:10]: The name of fields m2o, o2m, and m2m, need to include `_id` (m2o) or `_ids` (om2, m2m) at the end
In the case with `_ids` we need to include the string with the plural


`responsible_user_id = fields.Many2one("res.users")`
This comment applies to this file and the session file

File: open_academy/models/course.py [hunk 2]
```diff
@@ -5,5 +5,7 @@ class Course(models.Model):
...
+    sessions = fields.One2many("session", "course")
```
Comment (CarmenMiranda) [inline open_academy/models/course.py:11]:
```python
session_ids = fields.One2many(
    "session", "course_id",
    string="Sessions",
)
```

**Review by CarmenMiranda — CHANGES_REQUESTED** [review]

File: open_academy/models/session.py [hunk 3]
```diff
@@ -9,3 +9,6 @@ class Session(models.Model):
     start_date = fields.Date()
     duration = fields.Float()
     number_of_seats = fields.Integer()
+    instructor = fields.Many2one("res.partner")
```
Comment (andreaarce) [inline open_academy/models/session.py:12]: Please here apply the comments of @CarmenMiranda about the name of the field for m2o for the course and attendees fields too

**Comment by JesusValdez96** [comment]
Please review again @CarmenMiranda, @andreaarce

**Review by CarmenMiranda — APPROVED** [review]
Technically LGTM 👍🏼

**Pull request merged**